x-pack/plugins/lens/public/plugin.ts

```diff
@@ -389,12 +393,21 @@ export class LensPlugin {
     if (share) {
       this.locator = share.url.locators.create(new LensAppLocatorDefinition());
 
-      share.register(
-        downloadCsvShareProvider({
-          uiSettings: core.uiSettings,
-          formatFactoryFn: () => startServices().plugins.fieldFormats.deserialize,
-        })
-      );
+      const { getStartServices } = core;
+      const startServices$ = from(getStartServices());
+      startServices$.subscribe(([, { licensing }]) => {
+        licensing?.license$.subscribe((license) => {
+          const atLeastGold = license.hasAtLeast('gold');
+          return share.register(
+            downloadCsvShareProvider({
+              uiSettings: core.uiSettings,
+              formatFactoryFn: () => startServices().plugins.fieldFormats.deserialize,
+              atLeastGold,
+              isNewVersion: share.isNewVersion,
+            })
+          );
+        });
+      });
```

Comment:
I would find it a bit easier to read making `isAtLeastGold` a function to be called rather than having two nested `subscribe` calls:

```suggestion
      share.register(
        downloadCsvShareProvider({
          uiSettings: core.uiSettings,
          formatFactoryFn: () => startServices().plugins.fieldFormats.deserialize,
          atLeastGold: () => {
            let isGold = false;
            startServices()
              .plugins.licensing?.license$.pipe(take(1))
              .subscribe((license) => {
                isGold = license.hasAtLeast('gold');
              });
            return isGold;
          },
          isNewVersion: share.isNewVersion,
        })
      );
```

Reply:
`atLeastGold` needs to be changed in the `csv_download_provider` part accordingly.